compiler/src/iree/compiler/Codegen/LLVMGPU/ConvertToLLVM.cpp

```diff
@@ -44,6 +45,74 @@ LogicalResult verifyLLVMConversionCompatibility(ModuleOp moduleOp) {
   return compatible;
 }
 
+/// Returns the LLVM type of the global variable given the memref type `type`.
+static Type convertGlobalMemrefTypeToLLVM(MemRefType type,
+                                          LLVMTypeConverter &typeConverter) {
+  // LLVM type for a global memref will be a multi-dimension array. For
+  // declarations or uninitialized global memrefs, we can potentially flatten
+  // this to a 1D array. However, for memref.global's with an initial value,
+  // we do not intend to flatten the ElementsAttribute when going from std ->
+  // LLVM dialect, so the LLVM type needs to me a multi-dimension array.
+  Type elementType = typeConverter.convertType(type.getElementType());
+  Type arrayTy = elementType;
+  // Shape has the outermost dim at index 0, so need to walk it backwards
+  for (int64_t dim : llvm::reverse(type.getShape()))
+    arrayTy = LLVM::LLVMArrayType::get(arrayTy, dim);
+  return arrayTy;
+}
+
+/// Alias group information needs to be propagated to LLVM dialect, therefore we
+/// need our own lowering.
+struct GlobalMemrefOpLowering
```

Comment:
What alternative do you suggest? This is related to the discussion brought up by Ben. Eventually we will need our own lowering to LLVM global no matter what since we cannot represent those aliased memref at the memref dialect level. If we go with what Ben suggested then this will be replaced by our own lowering of the new op that will generate LLVM global. 
Can you give signal on whether you are in agreement with the suggested solution of adding a new iree op for that.

Reply:
I left a comment on the PR related to this.

Reply:
thanks I had missed that.